mmdet/core/bbox/match_costers/match_cost.py

```diff
@@ -0,0 +1,102 @@
+import torch
+
+from mmdet.core.bbox.iou_calculators import build_iou_calculator
+from mmdet.core.bbox.transforms import bbox_xyxy_to_cxcywh
+from .builder import MATCH_COST
+
+
+@MATCH_COST.register_module()
+class BBoxL1Cost(object):
+
+    def __init__(self, weight=1.):
+        self.weight = weight
+
+    def __call__(self, bbox_pred, gt_bboxes, factor):
+        """
+        Args:
+            bbox_pred (Tensor): Predicted boxes with normalized coordinates
+                (cx, cy, w, h), which are all in range [0, 1]. Shape
+                [num_query, 4].
+            gt_bboxes (Tensor): Ground truth boxes with unnormalized
+                coordinates (x1, y1, x2, y2). Shape [num_gt, 4].
+            factor (Tensor): Image size(whwh) tensor. Shape [4].
+        """
+        gt_bboxes_normalized = gt_bboxes / factor
+        # MODYFY: xyxy p1-norm in sparse-RCNN CODE which need to review
+        bbox_cost = torch.cdist(
```

Comment:
If you want to keep the `scale_factor`, maybe it is better to add the default value [1,1,1,1], and also normalize `bbox_pred` at the same time, or just delete it,  because  we can normalize `bbox_pred` and `gt_bboxes` with `whwh` outside of assign when we need(such as in sparse rcnn), because the normalization does not affect the value of ` iou` or `giou` loss